### PR TITLE
Adding new API version 2018-07-10 for Azure Site Recovery Services

### PR DESCRIPTION
**Pull request by TalluriAnusha**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 1]
```diff
@@ -0,0 +1,16929 @@
...
+          },
+          {
+            "name": "protectionContainerName",
+            "in": "path",
+            "description": "Protection container name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "migrationI
```
Comment (TalluriAnusha) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: InMageAzureV2DiskInputDetails(@10514 line) is the new class introduced.

Comment (TalluriAnusha) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: required flag is removed here for storageAccountId.

Comment (TalluriAnusha) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: type used to be string, is converted to InMageAzureV2DiskInputDetails.

Comment (TalluriAnusha) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: new input property added for disktype input at VM level.
Also removed userManagedDisks input property, which is obsolete.

Comment (TalluriAnusha) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: new class InMageAzureV2ManagedDiskDetails is introduced for GetVM/ListVM calls.

Comment (TalluriAnusha) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json]: new property added for retrieving the managed disk details.

**Comment by sriramvu** [comment]
why replicaDiskType in InMageAzureV2ManagedDiskDetails and diskType in InMageAzureV2DiskInputDetails are framed differently when both turns out to serve similar purpose - by giving supported values.

**Comment by TalluriAnusha** [comment]
InMageAzureV2ManagedDiskDetails is the Get object, so we've kept that as string instead of enum.

**Pull request merged**